In [1]:
import pandas as pd

# Crear DataFrame desde los datos
data = {
    'Bloque': list(range(1, 11)) * 3,
    'Tratamiento': ['Tratamiento 1']*10 + ['Tratamiento 2']*10 + ['Tratamiento 3']*10,
    'Semillas': [
        7, 8, 9, 10, 11, 8, 7, 8, 7, 8,
        9, 9, 9, 9, 12, 10, 8, 8, 9, 9,
        10, 10, 12, 12, 14, 9, 7, 7, 10, 10
    ]
}

df = pd.DataFrame(data)
df.head()


Bloque    Tratamiento  Semillas
0       1  Tratamiento 1         7
1       2  Tratamiento 1         8
2       3  Tratamiento 1         9
3       4  Tratamiento 1        10
4       5  Tratamiento 1        11

In [2]:
import statsmodels.api as sm
from statsmodels.formula.api import ols

# Modelo con bloques y tratamientos
modelo = ols('Semillas ~ C(Tratamiento) + C(Bloque)', data=df).fit()
anova_table = sm.stats.anova_lm(modelo, typ=2)
print(anova_table)


                sum_sq    df         F    PR(>F)
C(Tratamiento)    16.2   2.0  9.227848  0.001744
C(Bloque)         54.8   9.0  6.936709  0.000260
Residual          15.8  18.0       NaN       NaN


**PUNTO A**

**HIPOTESIS**

H₀: No hay diferencia en el número medio de semillas entre tratamientos.

H₁: Al menos un tratamiento tiene un efecto diferente sobre la producción de semillas.

**Resultados:**

**Tratamiento:**

F = 9.23

p-valor = 0.001744 (< 0.05)

**Bloque (Abeto):**

F = 6.94

p-valor = 0.000260 (< 0.05)

**Resultado o efecto del tratamiento:**

Como el p-valor es menor a 0.05, rechazamos la hipótesis nula.


Por lo tanto los tratamientos afectan significativamente la producción de semillas.

Por otro lado, se demuestra que no todos los tratemeintos son iguales, por lo cual se recomienda hacer una comparacion para ver cuales son mejor

In [4]:
import pingouin as pg  # Librería para análisis estadístico, similar a SPSS en Python

# Realizar prueba de comparaciones múltiples con corrección de Bonferroni
bonferroni = pg.pairwise_ttests(
    dv='Semillas',          # Variable dependiente: número de semillas
    between='Tratamiento',  # Factor de comparación: los tratamientos
    subject='Bloque',       # Bloque aleatorizado: cada árbol es un bloque
    padjust='bonf',         # Método de corrección: Bonferroni
    data=df                 # Conjunto de datos
)

# Mostrar solo las columnas clave del resultado
bonferroni[['A', 'B', 'T', 'p-unc', 'p-corr', 'p-adjust']]


c:\Users\$\AppData\Local\Programs\Python\Python313\Lib\site-packages\pingouin\pairwise.py:28: UserWarning: pairwise_ttests is deprecated, use pairwise_tests instead.
  warnings.warn("pairwise_ttests is deprecated, use pairwise_tests instead.", UserWarning)


A              B         T     p-unc    p-corr p-adjust
0  Tratamiento 1  Tratamiento 2 -1.622273  0.122129  0.366387     bonf
1  Tratamiento 1  Tratamiento 3 -2.223145  0.039252  0.117757     bonf
2  Tratamiento 2  Tratamiento 3 -1.156553  0.262575  0.787724     bonf

**PUNTO B**

Aunque en el punto anterior se pudo evidencias variaciones improtantes, entre los tratameintos, el tyest de boferroni, no tiene o no identifica diferencias estadisticas muy relevantes entro los pares

Esto puede ser tambien causado por Bonferroni, es muy consevador y puede darnos una disminucion en la eficacia estatidisca en muestra muy reducidas 
